.circleci/config.yml

```diff
@@ -0,0 +1,59 @@
+version: 2.1
+
+references:
+  images:
+    go: &GOLANG_IMAGE circleci/golang:latest
+  environments:
+    tmp: &TEST_RESULTS_PATH /tmp/test-results # path to where test results are saved
+
+# reusable 'executor' object for jobs
+executors:
+  go:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      - TEST_RESULTS: *TEST_RESULTS_PATH
+
+jobs:
+  go-test:
+    executor: go
+    steps:
+      - checkout
+      - run: mkdir -p $TEST_RESULTS
+
+      - restore_cache: # restore cache from dev-build job
+          keys:
+            - go-retryablehttp-modcache-v1-{{ checksum "go.mod" }}
+
+      - run: go mod download
+
+      # Save go module cache if the go.mod file has changed
+      - save_cache:
+          key: go-retryablehttp-modcache-v1-{{ checksum "go.mod" }}
+          paths:
+            - "/go/pkg/mod"
+
+      # check go fmt output because it does not report non-zero when there are fmt changes
+      - run:
+          name: check go fmt
+          command: |
+            files=$(go fmt ./...)
```

Comment:
Be careful here, as this will fail if there is a syntax error in the go code, but will place that error on stderr so you'll miss it in the condition below. You could either just redirect stderr to capture it (`files=$(go fmt ./... 2>&1)`) or switch on automatic error detection in bash (see above).